scope3ai/tracers/huggingface/image_to_image.py

```diff
@@ -0,0 +1,115 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Optional
+
+import tiktoken
+from aiohttp import ClientResponse
+from huggingface_hub import ImageToImageOutput as _ImageToImageOutput
+from huggingface_hub import InferenceClient, AsyncInferenceClient  # type: ignore[import-untyped]
+from requests import Response
+
+from scope3ai.api.types import Scope3AIContext, Model, ImpactRow
+from scope3ai.api.typesgen import Task
+from scope3ai.constants import PROVIDERS
+from scope3ai.lib import Scope3AI
+from scope3ai.response_interceptor.aiohttp_interceptor import aiohttp_response_capture
+from scope3ai.response_interceptor.requests_interceptor import requests_response_capture
+
+PROVIDER = PROVIDERS.HUGGINGFACE_HUB.value
+
+
+@dataclass
+class ImageToImageOutput(_ImageToImageOutput):
+    scope3ai: Optional[Scope3AIContext] = None
+
+
+def huggingface_image_to_image_wrapper_non_stream(
+    wrapped: Callable, instance: InferenceClient, args: Any, kwargs: Any
+) -> ImageToImageOutput:
+    http_response: Response | None = None
+    with requests_response_capture() as responses:
+        response = wrapped(*args, **kwargs)
+        http_responses = responses.get()
+        if len(http_responses) > 0:
+            http_response = http_responses[-1]
+
+    model = kwargs.get("model") or instance.get_recommended_model("text-to-speech")
+    compute_time = http_response.headers.get("x-compute-time")
+    input_tokens = http_response.headers.get("x-compute-characters")
+    # TODO: Load image in memory?
```

Comment:
This will break if http_response is None for any reason (wrapper does not work or such)